stan/math/prim/scal/prob/inv_chi_square_lccdf.hpp

```diff
@@ -46,7 +46,8 @@ namespace stan {
       typedef typename stan::partials_return_type<T_y, T_dof>::type
         T_partials_return;
 
-      if ( !( stan::length(y) && stan::length(nu) ) ) return 0.0;
+      if (!(stan::length(y) && stan::length(nu)))
+        return 0.0;
```

Comment:
We don't need to use `0.0` --- `0` is sufficient.  The assembly code winds up being the same as the cast happens statically.  I'm not suggesting you need to go through and simplify all the redundnat `.0` number suffixes in the code.